src/pf-tooltip/pf-tooltip.component.js

```diff
@@ -0,0 +1,304 @@
+    }
+  }
+
+  /**
+   * Get the tooltip container element
+   *
+   * @returns {string} The container element
+   */
+  get container () {
+    return this._container;
+  }
+
+  /**
+   * Set the tooltip container element
+   *
+   * @param {string} value The container element
+   */
+  set container (value) {
+    if (this._container !== value) {
+      this._container = value;
+      this.setAttribute('container', value);
+    }
+  }
+
+  /**
+   * Get the animation duration
+   *
+   * @returns {string} The animation duration
+   */
+  get duration () {
+    return this._duration;
+  }
+
+  /**
+   * Set the animation duration
+   *
+   * @param {string} value The animation duration
+   */
+  set duration (value) {
+    if (this._duration !== value) {
+      this._duration = value;
+      this.setAttribute('duration', value);
+    }
+  }
+
+  /**
+   * Get the animation delay
+   *
+   * @returns {string} The animation delay
+   */
+  get delay () {
+    return this._duration;
+  }
+
+  /**
+   * Set the animation delay
+   *
+   * @param {string} value The animation delay
+   */
+  set delay (value) {
+    if (this._delay !== value) {
+      this._delay = value;
+      this.setAttribute('delay', value);
+    }
+  }
+
+  /**
+   * Get the placement position
+   *
+   * @returns {string} The placement position left, top, bottom, right
+   */
+  get placement () {
+    return this._placement;
+  }
+
+  /**
+   * Set placement position
+   *
+   * @param {string} value The placement position left, top, bottom, right
+   */
+  set placement (value) {
+    if (this._placement !== value) {
+      this._placement = value;
+      this.setAttribute('placement', value);
+    }
+  }
+
+  /**
+   * Get the target element
+   *
+   * @returns {string} The target element
+   */
+  get target () {
+    return this._target;
+  }
+
+  /**
+   * Set target element
+   *
+   * @param {string} value The target element
+   */
+  set target (value) {
+    if (this._target !== value) {
+      this._target = value;
+      this.setAttribute('target', value);
+    }
+  }
+
+  /**
+   * The open event handler
+   * @param e
+   */
+  open (e) {
+    clearTimeout(this._timer);
+    this._timer = setTimeout( () => {
+      if (this.tooltip === null) {
+        this.createTooltip();
+        this.styleTooltip();
+        this.showTooltip();
+      }
+    }, 20 );
+  }
+
+  /**
+   * The close event handler
+   * @param e
+   */
+  close (e) {
+    clearTimeout(this._timer);
+    this._timer = setTimeout( () => {
+      if (this.tooltip && this.tooltip !== null) {
+        pfUtil.removeClass(this.tooltip, 'in');
+        setTimeout(() => {
+          this.removeTooltip();
+        }, this._duration);
+      }
+    }, this._delay + this._duration);
+  }
+
+  /**
+   * Removes the tooltip
+   */
+  removeTooltip () {
+    this.tooltip && this.container.removeChild(this.tooltip);
+    this.tooltip = null;
+  }
+
+  /**
+   * Creates the tooltip
+   */
+  createTooltip () {
```

Comment:
Should createTooltip be private (i.e., with an underscore)?

Likewise, removeTooltip, open, close, etc. seem like they could possibly be private functions?